pennylane/estimator/resource_tracking.py

```diff
@@ -0,0 +1,379 @@
+
+    for action in resource_decomp:
+        if isinstance(action, GateCount):
+            _update_counts_from_compressed_res_op(
+                action.gate,
+                gate_counts_dict,
+                wire_mngr=wire_mngr,
+                scalar=scalar * action.count,
+                gate_set=gate_set,
+                config=config,
+            )
+            continue
+
+        if isinstance(action, Allocate):
+            if qubit_alloc_sum != 0 and scalar > 1:
+                wire_mngr.grab_zeroed(action.num_wires * scalar)
+            else:
+                wire_mngr.grab_zeroed(action.num_wires)
+        if isinstance(action, Deallocate):
+            if qubit_alloc_sum != 0 and scalar > 1:
+                wire_mngr.free_wires(action.num_wires * scalar)
+            else:
+                wire_mngr.free_wires(action.num_wires)
+
+    return
+
+
+def _sum_allocated_wires(decomp):
+    """Sum together the allocated and released wires in a decomposition."""
+    s = 0
+    for action in decomp:
+        if isinstance(action, Allocate):
+            s += action.num_wires
+        if isinstance(action, Deallocate):
+            s -= action.num_wires
+    return s
+
+
+@QueuingManager.stop_recording()
+def _ops_to_compressed_reps(
+    ops: Iterable[Operation | ResourceOperator],
+) -> list[CompressedResourceOp]:
+    """Convert the sequence of operations to a list of compressed resource ops.
+
+    Args:
+        ops (Iterable[Union[Operation, ResourceOperator]]): set of operations to convert
+
+    Returns:
+        List[CompressedResourceOp]: set of converted compressed resource ops
+    """
+    cmp_rep_ops = []
+    for op in ops:  # We are skipping measurement processes here.
+        if isinstance(op, ResourceOperator):
+            cmp_rep_ops.append(op.resource_rep_from_op())
+
+        if isinstance(op, Operation):  # map: op --> res_op, then: res_op --> cmprsd_res_op
+            cmp_rep_ops.append(map_to_resource_op(op).resource_rep_from_op())
+
+    return cmp_rep_ops
+
+
+def _get_decomposition(
+    cp_rep: CompressedResourceOp, config: ResourceConfig
+) -> tuple[Callable, dict]:
+    """
+    Selects the appropriate decomposition function and kwargs from a config object.
+
+    This helper function centralizes the logic for choosing a decomposition,
+    handling standard, custom, and symbolic operator rules using a mapping.
+
+    Args:
+        cp_rep (CompressedResourceOp): The operator to find the decomposition for.
+        config (ResourceConfig): The configuration object containing decomposition rules.
+
+    Returns:
+        A tuple containing the decomposition function and its associated kwargs.
+    """
+    op_type = cp_rep.op_type
+    _SYMBOLIC_DECOMP_MAP = {
+        # TODO: Uncomment this when symbolic resource operators are merged.
+        # ResourceAdjoint: "_adj_custom_decomps",
+        # ResourceControlled: "_ctrl_custom_decomps",
+        # ResourcePow: "_pow_custom_decomps",
```

Comment:
[optional] Not sure if we should keep these here or add them when the relevant features are merged.

Reply:
Either is okay by me. I think it would be easier to just uncomment out this block as needed.